packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessForm.tsx

```diff
@@ -0,0 +1,244 @@
+  const projectCode = project?.code;
+
+  const queryResult = useCountryAccessList();
+  const countries: TupaiaWebCountryAccessListRequest.ResBody = queryResult.data ?? [];
+  const accessListIsLoading = queryResult.isLoading;
+
+  const { mutate: requestCountryAccess, isLoading: requestIsLoading } = useRequestProjectAccess({
+    onError: error =>
+      errorToast(error?.message ?? 'Sorry, couldn’t submit your request. Please try again'),
+    onSettled: () => reset(),
+    onSuccess: response => {
+      successToast(response.message);
+      rerenderCountryChecklist();
+    },
+  });
+
+  /**
+   * When React Hook Form resets the form state, boxes get correctly unchecked, but the MUI
+   * component fails to reflect this until its next render (usually when the user interacts with it,
+   * which makes the checkbox feel broken). setCountryCheckList serves merely as an "automatic
+   * kicking machine" to force a re-render.
+   */
+  const [countryChecklistKey, setCountryChecklistKey] = useState(Date.now());
+
+  /**
+   * HACK: Force a fresh render of the country checklist. Use when a MUI checkbox appears checked,
+   * but is actually unchecked (or vice versa).
+   */
+  const rerenderCountryChecklist = () => setCountryChecklistKey(Date.now());
+
+  const sizeClassIsMdOrLarger = useMediaQuery(theme.breakpoints.up('sm'));
+
+  const formContext = useForm<RequestCountryAccessFormFields>({
+    defaultValues: {
+      countryIds: [],
+      reasonForAccess: '',
+    } as RequestCountryAccessFormFields,
+    mode: 'onChange',
+  });
+  const {
+    formState: { isSubmitting, isValidating, isValid },
+    handleSubmit,
+    register,
+    reset,
+  } = formContext;
+
+  const applicableCountries =
+    countries?.filter((country: TupaiaWebCountryAccessListRequest.CountryAccess) =>
+      project?.names?.includes(country.name),
+    ) ?? [];
+
+  const formIsNotSubmissible =
+    !project || isValidating || !isValid || isSubmitting || accessListIsLoading || requestIsLoading;
+
+  function onSubmit(formData: RequestCountryAccessFormFields) {
+    requestCountryAccess({
+      entityIds: formData.countryIds,
+      message: formData.reasonForAccess,
+      projectCode: projectCode,
+    });
+  }
+
+  return (
+    <StyledForm formContext={formContext} onSubmit={handleSubmit(onSubmit)}>
+      <StyledFieldset>
+        <CountryChecklistWrapper>
+          <StyledFormLabel>Select countries</StyledFormLabel>
+          <CountryChecklist
+            disabled={!project || isSubmitting || requestIsLoading}
+            key={countryChecklistKey}
+          >
+            {applicableCountries.map(({ id, name, hasAccess, accessRequests }) => {
+              const hasRequestedAccess = accessRequests.includes(ensure(projectCode));
+              const getTooltip = () => {
+                if (hasAccess) return 'You already have access';
+                if (hasRequestedAccess) return 'Approval in progress';
+              };
+
```

Comment:
I think `getTooltip` could be moved outside of this loop, and made to accept `hasAccess` and `hasRequestedAccess` params, since it's quite a nice generic function.

I would also add a line here `const tooltip = getTooltip()` to match the pattern we use elsewhere, and possibly also reduce re-rendering from an inline function (although I'm not sure of that part)